modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java

```diff
@@ -768,10 +767,6 @@ public static void compressDeletedEntries(Map<String, ?> prefixMap) {
                 if (map.containsKey(NamedListNode.NAME) && map.get(NamedListNode.NAME) == null) {
                     entry.setValue(null);
                 }
-            } else if (value == null) {
```

Comment:
Why did you remove this clause but not the other?

Reply:
Because previous clause is just a deletion of a named list element, but this particular clause is related to a different case that's now handled in another method